ledger-tool/src/main.rs

```diff
@@ -1398,702 +1422,756 @@ fn main() {
-                    .takes_value(true)
-                    .value_name("ENDING_SLOT")
-                    .help("Ending slot for minimized snapshot creation")
-            )
-            .arg(
-                Arg::with_name("snapshot_archive_format")
-                    .long("snapshot-archive-format")
-                    .possible_values(SUPPORTED_ARCHIVE_COMPRESSION)
-                    .default_value(DEFAULT_ARCHIVE_COMPRESSION)
-                    .value_name("ARCHIVE_TYPE")
-                    .takes_value(true)
-                    .help("Snapshot archive format to use.")
-                    .conflicts_with("no_snapshot")
-            )
-        ).subcommand(
+                .about("Create a new ledger snapshot")
+                .arg(&no_snapshot_arg)
+                .arg(&account_paths_arg)
+                .arg(&accounts_hash_cache_path_arg)
+                .arg(&accounts_index_bins)
+                .arg(&accounts_index_limit)
+                .arg(&disable_disk_index)
+                .arg(&accountsdb_verify_refcounts)
+                .arg(&accounts_db_test_skip_rewrites_but_include_in_bank_hash)
+                .arg(&accounts_db_skip_initial_hash_calc_arg)
+                .arg(&accountsdb_skip_shrink)
+                .arg(&ancient_append_vecs)
+                .arg(&hard_forks_arg)
+                .arg(&max_genesis_archive_unpacked_size_arg)
+                .arg(&snapshot_version_arg)
+                .arg(&maximum_full_snapshot_archives_to_retain)
+                .arg(&maximum_incremental_snapshot_archives_to_retain)
+                .arg(&geyser_plugin_args)
+                .arg(&use_snapshot_archives_at_startup)
+                .arg(
+                    Arg::with_name("snapshot_slot")
+                        .index(1)
+                        .value_name("SLOT")
+                        .validator(|value| {
+                            if value.parse::<Slot>().is_ok() || value == "ROOT" {
+                                Ok(())
+                            } else {
+                                Err(format!(
+                                    "Unable to parse as a number or the keyword ROOT, provided: \
+                                     {value}"
+                                ))
+                            }
+                        })
+                        .takes_value(true)
+                        .help(
+                            "Slot at which to create the snapshot; accepts keyword ROOT for the \
+                             highest root",
+                        ),
+                )
+                .arg(
+                    Arg::with_name("output_directory")
+                        .index(2)
+                        .value_name("DIR")
+                        .takes_value(true)
+                        .help(
+                            "Output directory for the snapshot [default: --snapshot-archive-path \
+                             if present else --ledger directory]",
```

Comment:
```suggestion
                            "Output directory for the snapshot \
                            [default: --snapshot-archive-path if present else --ledger directory]",
```
If you're keen to match what I did in CLI, and this is short enough to not break fmt.